internal/table/client.go

```diff
@@ -257,6 +261,47 @@ func (c *Client) DoTx(ctx context.Context, op table.TxOperation, opts ...table.O
 	}, config.RetryOptions...)
 }
 
+func (c *Client) BulkUpsert(
+	ctx context.Context,
+	tableName string,
+	data table.BulkUpsertData,
+	opts ...table.Option,
+) (finalErr error) {
+	if c == nil {
+		return xerrors.WithStackTrace(errNilClient)
+	}
+
+	if c.isClosed() {
+		return xerrors.WithStackTrace(errClosedClient)
+	}
+
+	config := c.retryOptions(opts...)
+
+	attempts, onDone := 0, trace.TableOnBulkUpsert(config.Trace, &ctx,
+		stack.FunctionID("github.com/ydb-platform/ydb-go-sdk/v3/internal/table.(*Client).BulkUpsert"),
+		config.Label,
+	)
+	defer func() {
+		onDone(attempts, finalErr)
+	}()
+
+	return retryBackoff(ctx, c.pool, func(ctx context.Context, s table.Session) (err error) {
```

Comment:
In this place need to use retry.Retry with flag Idempotent - because no need to use session pool